### ISSUE TITLE
new branch for aiida-core 2.0 support

### ISSUE THREAD

**Issue by ltalirz** [aiidalab/aiidalab-docker-stack#240]
Given that [aiida-dynamic-workflows](https://github.com/microsoft/aiida-dynamic-workflows) is already based on the aiida-core `develop` branch, I'm finding myself in the need of an AiiDAlab environment that supports AiiDA 2.0.

I'll have a quick stab at updating the Dockerfile and will be pushing this to a separate `aiida-2` branch.
We can use this issue to keep track of other packages that need updating

- [ ] aiidalab-widgets-base: needs to change aiida-core dependency

**Comment by csadorf** [comment]
@ltalirz Thanks for taking the initiative. I am going to assign you for now since you already started working on this, but please feel free to tab me in for any of these tasks.

**Comment by ltalirz** [comment]
cheers; the aiida-2 branch now exists and I'm using it https://github.com/aiidalab/aiidalab-docker-stack/pull/241
Feel free to use this branch as well and make further updates/improvements

I think you can move ahead with merging https://github.com/aiidalab/aiidalab/pull/269 and https://github.com/aiidalab/aiidalab-eln/pull/34

**Comment by csadorf** [comment]
Note: This effort is currently blocked, because the latest aiidateam/aiida-core are based on the latest aiidateam/prerequisites images that use a newer postgres version which – without migration – will break existing user installations.

See also: https://github.com/aiidalab/aiidalab-docker-stack/issues/25

**Comment by ltalirz** [comment]
If you feel that this is the main issue holding you back and migration is too difficult, you can also downgrade postgres (the main reason for the postgres upgrade was to upgrade the OS).

However, if you do migrate, and postgres upgrades are so diffficult, then perhaps consider going to [ubuntu 22.04 and postgres 14](https://packages.ubuntu.com/jammy/database/postgresql).

**Comment by csadorf** [comment]
> If you feel that this is the main issue holding you back and migration is too difficult, you can also downgrade postgres (the main reason for the postgres upgrade was to upgrade the OS).

Yes, thank you for the suggestion. We have already decided (in today's dev meeting) to downgrade and slightly postpone the migration to unblock this issue.

**Comment by unkcpz** [comment]
After this is done, I think most apps in app store would incompatible since they are specifically set to dependent on `aiida-core~=1.0`. We need to disallow the user to install the old versions of these apps until they have aiida-core 2.0 supported.

**Comment by csadorf** [comment]
> After this is done, I think most apps in app store would incompatible since they are specifically set to dependent on `aiida-core~=1.0`. We need to disallow the user to install the old versions of these apps until they have aiida-core 2.0 supported.

@unkcpz We could modify the [`is_comaptible()`](https://github.com/aiidalab/aiidalab/blob/04f295b6eb8aee851a4eb3f538369dd441ce3ea9/aiidalab/app.py#L258-L259) function too look at the constraints instead of the current environment.

**Comment by unkcpz** [comment]
> @unkcpz We could modify the [is_comaptible()](https://github.com/aiidalab/aiidalab/blob/04f295b6eb8aee851a4eb3f538369dd441ce3ea9/aiidalab/app.py#L258-L259) function too look at the constraints instead of the current environment.

I open an issue for the compatible check in `aiidalab` https://github.com/aiidalab/aiidalab/issues/318. So this issue can be closed I think.
